### ISSUE TITLE
Focus input AFTER modal is closed

### ISSUE THREAD

**Issue by AnthoniG** [reactstrap/reactstrap#420]
Hi,
Loving the reactstrap components. Using them in my new project.
However I am having trouble with the modal box.
I have a form that needs to be filled. Then the user clicks a button and a Modal opens to fill in some extra information. Upon closing the Modal I need one of the input type="text" fields to be focused. However no matter what I do it's not working.

Tried using refs, as per React docs, but the object brought back does not understand .focus() method. How do I get the underlying component that is wrapped ?

Regards
Anthoni

**Comment by TheSharpieOne** [comment]
Because reactstrap wraps the raw input in a component, the `ref` will give you back the wrapping component, not the raw input. The `ref` prop is special, reactstrap cannot directly access it to pass it through. However, reactstrap does have an extra prop, `getRef` which you can use just like `ref`. `getRef` will be passed to the raw input, allowing you to get the `ref` of that input.

tl;dr: use `getRef` instead of `ref` for reactstrap's `<Input>`.

**Comment by AnthoniG** [comment]
Hi @TheSharpieOne I am using Typescript and when I put in getRef I get an error saying that it's not allowed.

This is how I am currently using ref:
`ref={(input) => { this.queryInput = input; }}`

**Comment by TheSharpieOne** [comment]
It's allowed I'm JSX, and pure, plain old JavaScript if you don't use JSX. I don't know too much about typescript, but there should be a way to tell it to get out of your way let you do it.

**Comment by AnthoniG** [comment]
hhhm, I might have to edit the .d.ts file myself a little bit 😃

**Comment by TheSharpieOne** [comment]
I am closing this as it seem like the question have been answered. If you believe this is not the case, please comment and I will reopen the issue.

**Comment by NathanCH** [comment]
Incase anyone gets here from Google. The above solution is outdated. This is how you get ref in Reactstrap <7.1

```
<Input ref={n => this.node = n} />
```

Change to

```
<Input innerRef={n => this.node = n} />
```
